src/vec-final.md

```diff
@@ -304,5 +304,77 @@ impl<'a, T> Drop for Drain<'a, T> {
     }
 }
 
-# fn main() {}
+# fn main() {
+#     tests::create_push_pop();
+#     tests::iter_test();
+#     tests::test_drain();
+#     tests::test_zst();
+#     println!("All tests finished OK");
+# }
+
+# mod tests {
+#     use super::*;
+#     pub fn create_push_pop() {
+#         let mut v = Vec::new();
+#         v.push(1);
+#         assert_eq!(1, v.len());
+#         assert_eq!(1, v[0]);
+#         for i in v.iter_mut() {
+#             *i += 1;
+#         }
+#         v.insert(0, 1);
```

Comment:
Maybe insert a different value here so we can distinguish them, making sure the right one gets indeed popped?